File: app/src/main/java/com/bulingzhuang/aimd/view/activity/EditorActivity.java
package com.bulingzhuang.aimd.view.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.transition.Fade;
import android.transition.TransitionManager;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.bulingzhuang.aimd.R;
import com.bulingzhuang.aimd.entity.ModuleTextEntity;
import com.bulingzhuang.aimd.entity.ModuleTitleEntity;
import com.bulingzhuang.aimd.utils.Tools;
import com.bulingzhuang.aimd.utils.UploadModuleUtil;
import com.google.gson.Gson;

import org.simple.eventbus.EventBus;
import org.simple.eventbus.Subscriber;

import java.util.ArrayList;

import butterknife.Bind;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class EditorActivity extends AppCompatActivity {


    @Bind(R.id.tv_add)
    TextView tvAdd;
    @Bind(R.id.ll_add)
    LinearLayout llAdd;
    @Bind(R.id.ll_content)
    LinearLayout llContent;
    @Bind(R.id.iv_del)
    ImageView ivDel;
    @Bind(R.id.iv_finish)
    ImageView ivFinish;
    @Bind(R.id.ll_parent)
    LinearLayout llParent;
    private Gson mGson;
    private ArrayList<String> mContentArray;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_editor);
        ButterKnife.bind(this);
        EventBus.getDefault().register(this);
        Tools.changeFont(tvAdd);
        mGson = new Gson();
        mContentArray = new ArrayList<>();
    }

    @Override
    protected void onDestroy() {
        EventBus.getDefault().unregister(this);
        ButterKnife.unbind(this);
        super.onDestroy();
    }

    @Override
    public void onBackPressed() {
        TransitionManager.beginDelayedTransition(llAdd, new Fade());
        llAdd.setVisibility(View.INVISIBLE);
        super.onBackPressed();
    }

    @OnClick({R.id.iv_del, R.id.ll_add, R.id.iv_finish})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.iv_del:
                onBackPressed();
                break;
            case R.id.ll_add:
                Intent intent = new Intent(this, ModuleTypeActivity.class);
                intent.putStringArrayListExtra("contentArray", mContentArray);
                intent.putExtra("showTitleModule", llContent.getChildCount() == 0);
                startActivity(intent);
                break;
            case R.id.iv_finish:
                Tools.showSnackBar(this, "完成", llParent);
                break;
        }
    }

    @Subscriber(tag = "update_module_text")
    private void updateModuleText(String jsonData) {
        Tools.showLogE("Text编辑完成的内容：" + jsonData);
        ModuleTextEntity moduleTextData = mGson.fromJson(jsonData, ModuleTextEntity.class);
        UploadModuleUtil.uploadTextModule(this, llContent, moduleTextData);
        mContentArray.add(jsonData);
    }

    @Subscriber(tag = "update_module_title")
    private void updateModuleTitle(String jsonData) {
        Tools.showLogE("Title编辑完成的内容：" + jsonData);
        ModuleTitleEntity moduleTitleData = mGson.fromJson(jsonData, ModuleTitleEntity.class);
        UploadModuleUtil.uploadTitleModule(this, llContent, moduleTitleData);
        mContentArray.add(jsonData);
    }
}

File: app/src/main/java/com/bulingzhuang/aimd/view/activity/EditorActivity.kt
package com.bulingzhuang.aimd.view.activity

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.transition.Fade
import android.transition.TransitionManager
import android.view.View

import com.bulingzhuang.aimd.R
import com.bulingzhuang.aimd.entity.ModuleTextEntity
import com.bulingzhuang.aimd.entity.ModuleTitleEntity
import com.bulingzhuang.aimd.utils.Tools
import com.bulingzhuang.aimd.utils.UploadModuleUtil
import com.google.gson.Gson

import org.simple.eventbus.EventBus
import org.simple.eventbus.Subscriber

import java.util.ArrayList

import kotlinx.android.synthetic.main.activity_editor.*

class EditorActivity : AppCompatActivity(), View.OnClickListener {

    private var gson: Gson? = null
    private var mContentArray: ArrayList<String>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_editor)
        EventBus.getDefault().register(this)
        setViewsOnClickListener(iv_del,ll_add,iv_finish)
        Tools.changeFont(tv_add)
        gson = Gson()
        mContentArray = ArrayList<String>()
    }

    override fun onDestroy() {
        EventBus.getDefault().unregister(this)
        super.onDestroy()
    }

    override fun onBackPressed() {
        TransitionManager.beginDelayedTransition(ll_add, Fade())
        ll_add.visibility = View.INVISIBLE
        super.onBackPressed()
    }

    private fun setViewsOnClickListener(vararg views: View) {
        for (view in views) {
            view.setOnClickListener(this)
        }
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.iv_del -> onBackPressed()
            R.id.ll_add -> {
                val intent = Intent(this, ModuleTypeActivity::class.java)
                intent.putStringArrayListExtra("contentArray", mContentArray)
                intent.putExtra("showTitleModule", ll_content.childCount == 0)
                startActivity(intent)
            }
            R.id.iv_finish -> Tools.showSnackBar(this, "完成", ll_parent)
        }
    }

    @Subscriber(tag = "update_module_text")
    private fun updateModuleText(jsonData: String) {
        Tools.showLogE("Text编辑完成的内容：" + jsonData)
        val moduleTextData = gson!!.fromJson(jsonData, ModuleTextEntity::class.java)
        UploadModuleUtil.uploadTextModule(this, ll_content, moduleTextData)
        mContentArray!!.add(jsonData)
    }

    @Subscriber(tag = "update_module_title")
    private fun updateModuleTitle(jsonData: String) {
        Tools.showLogE("Title编辑完成的内容：" + jsonData)
        val moduleTitleData = gson!!.fromJson(jsonData, ModuleTitleEntity::class.java)
        UploadModuleUtil.uploadTitleModule(this, ll_content, moduleTitleData)
        mContentArray!!.add(jsonData)
    }
}
